src/settings/groups/GroupDetailsView.ts

```diff
@@ -0,0 +1,276 @@
+import m, {ChildArray, Children} from "mithril"
+import {Dialog} from "../../gui/base/Dialog.js"
+import {formatDateWithMonth, formatStorageSize} from "../../misc/Formatter.js"
+import {lang} from "../../misc/LanguageViewModel.js"
+import {getFirstOrThrow, neverNull} from "@tutao/tutanota-utils"
+import {GroupType} from "../../api/common/TutanotaConstants.js"
+import type {TableAttrs} from "../../gui/base/Table.js"
+import {ColumnWidth, Table, TableLineAttrs} from "../../gui/base/Table.js"
+import {logins} from "../../api/main/LoginController.js"
+import {Icons} from "../../gui/base/icons/Icons.js"
+import {showProgressDialog} from "../../gui/dialogs/ProgressDialog.js"
+import type {EntityUpdateData} from "../../api/main/EventController.js"
+import {TextField} from "../../gui/base/TextField.js"
+import type {DropDownSelectorAttrs} from "../../gui/base/DropDownSelector.js"
+import {DropDownSelector} from "../../gui/base/DropDownSelector.js"
+import type {UpdatableSettingsDetailsViewer} from "../SettingsView.js"
+import {locator} from "../../api/main/MainLocator.js"
+import {assertMainOrNode} from "../../api/common/Env.js"
+import {IconButton, IconButtonAttrs} from "../../gui/base/IconButton.js"
+import {ButtonSize} from "../../gui/base/ButtonSize.js"
+import {GroupDetailsModel} from "./GroupDetailsModel.js"
+
+assertMainOrNode()
+
+export class GroupDetailsView implements UpdatableSettingsDetailsViewer {
+	constructor(
+		private readonly model: GroupDetailsModel,
+	) {
+	}
+
+	/**
+	 * render the header that tells us what type of group we have here
+	 * @private
+	 */
+	private renderHeader(): Children {
+		return m(".h4.mt-l", getGroupTypeDisplayName(this.model.getGroupType()))
+	}
+
+	renderView(): Children {
+		return m("#user-viewer.fill-absolute.scroll.plr-l", [
+			this.renderHeader(),
+			...this.renderCommonInfo(),
```

Comment:
you don't really need to spread it I think? I'd just make it return `Children` like usual and would dump it there and let Mithril flatten it. The only reason not to would be if we had keys.